### DISCUSSION TITLE
Methods Bug?

### DISCUSSION THREAD

**Discussion by su-aac** [UNC-Project-Open-AAC/OS-DPI#212] [Help]
Hello,

I'd like to ask about the Methods feature. I encountered an unusual behavior when I activated all three methods: mouse, switch, and pointer dwell. While using the switch method, I pressed space to select the "like" icon, for instance, and did not press enter yet. When I moved my mouse to another icon, let's say "finished," and stayed there for a while with the pointer dwell method activated, the "like" icon remained selected instead of switching to "finished." I'm curious to know if this is a bug or if it's an intentional setting.

Here is a screen recording of the issue. I hope you can see it from your end.
https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/7328d222-e821-4f5d-a8b8-8d77a1a091f8

Thank you!
Hsiao-Ting

**Comment by gbishop** [comment]
Sounds like a bug. I've never tested the interaction between the various methods.

**Comment by su-aac** [comment]
Here is the exported folder of the design.
[Mandarin_grid_Methods.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/12114433/Mandarin_grid_Methods.zip)

There are a couple more issues that appear to be bugs. When I press the Home key to switch from the English grid back to the Mandarin grid, it doesn't fully switch and seems to get stuck. Additionally, while on the Mandarin grid page, when I press an icon, it displays 2 words on the display bar instead of one. Please help me. Thank you!!

Here are the screen recordings for the issues.
https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123387291/012af903-340e-4ef4-86a3-b5f9c70d18e8

Please help me take a look. Thank you!!

**Comment by gbishop** [comment]
Several small issues in your design. 

First, the `Pointer Dwell` method includes a handler for `Pointer down` and the `Mouse` method also includes one. They both fire on `Pointer down`, that's why you get two words in the display. I deleted the `Pointer down` handler from the `Pointer Dwell` method to eliminate that issue. This is also why you had problems with the Home button. It was firing twice.

You may ask why was the `Pointer down` Handler in the `Pointer Dwell` Method? Simply because I found it convenient when testing `Pointer Dwell` to be able to click in the interface and it is harmless unless you intended to also activate other Methods using `Pointer down`.

It might be better to create yourself a new Method that merges the parts of the various Methods you want to use together than relying on them to properly interoperate in all situations.

You have an unconfigured Timer Handler in both `2 switch` and `Mouse`. I deleted those. What was your intention for them?

I have attached my modified version. If you would like to meet to create a new uber Method for your project, I'll be glad to help.

[Mandarin_grid_Methods_gb.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/12124823/Mandarin_grid_Methods_gb.zip)

**Reply by su-aac** [reply]
Hi Gary,

Thank you for your prompt and detailed feedback on the design. I appreciate your efforts in identifying the issues and providing solutions. I've reviewed the modified version you attached, and it looks great.
For the unconfigured Timer Handlers in 2 switch and Mouse methods, I was trying to use the methods for the first time and not fully understanding what all of the functions do interactively. I appreciate you removing them.
Your suggestion to create a new Method for the design is a great idea! I'll keep this in mind when designing.
Thanks again for your help!

Best regards,
Hsiao-Ting
